kaldo/controllers/displacement.py

```
@@ -9,6 +9,286 @@
+            for j in range(n_unit):
+                jp = atom_map[k, j]
+                col_shift_j = cell_shifts[k, j] - cell_shifts[k, i]
+                g_new_j   = grid @ R_int.T + col_shift_j[np.newaxis, :]
+                g_new_j_w = g_new_j % supercell_shape[np.newaxis, :]
+                l_j_new   = (g_new_j_w[:, 0] * ny * nz
+                             + g_new_j_w[:, 1] * nz
+                             + g_new_j_w[:, 2])
+
+                base_orig2 = i  * n_replicas * n_unit + j
+                base_new2  = ip * n_replicas * n_unit + jp
+                for l in range(n_replicas):
```

Ruff (0.15.12)

[error] Ambiguous variable name: `l`

(E741)